### PR TITLE
docs: update api specs

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

**Comment by haxxmaxx** [comment]
Do we ever define what a stringExpression is?

File: spec-configs/props.conf.js [hunk 1]
```diff
@@ -20,17 +20,35 @@ module.exports = {
...
+        url: 'https://qlik.dev/apis/json-rpc/qix/schemas/#%23%2Fdefinitions%2Fschemas%2Fentries%2FNxHypercubeMode',
+      },
+      NxAttrExprDef: {
+        url: 'https://qlik.dev/apis/json-rpc/qix/schemas#%23%2Fdefinitions%2Fschemas%2Fentries%2FNxAttrExprDef',
+      },
+      NxInlineDimensionDef: {
+        url: 'https://qlik.dev/apis/json-rpc/qix/schemas#%23%2Fdefinitions%2Fschemas%2Fentries%2FNxInlineDimensionDef',
+      },
+      NxInlineMeasureDef: {
+        url: 'https://qlik.dev/apis/json-rpc/qix/schemas#%23%2Fdefinitions%2Fschemas%2Fentries%2FNxInlineMeasureDef',
```
Comment (haxxmaxx) [inline spec-configs/props.conf.js:51]: Are all of these used?

Comment (LiKang6688) [reply spec-configs/props.conf.js:51]: Yes, when you run `yarn spec`, you will find that those types are missing

Comment (haxxmaxx) [reply spec-configs/props.conf.js:51]: ok 👍

File: src/object-properties.js [hunk 2]
```diff
@@ -1,29 +1,32 @@
...
     /** @type {MeasureProperties[]} */
     qMeasures: [],
+    /** @type {schemasNxHypercubeMode} */
     qMode: 'S',
     /** @type {boolean} */
     qSuppressZero: false,
     /** @type {boolean} */
     qSuppressMissing: true,
-    /** @type {Array} */
+    /** @type {Array<integer<int32>>} */
```
Comment (haxxmaxx) [inline src/object-properties.js:27]: isn't int enough?

Comment (LiKang6688) [reply src/object-properties.js:27]: I copied from somewhere else without thinking about it. Let me fix it.

Comment (haxxmaxx) [reply src/object-properties.js]: I mean why int32? Oddly specific for being js

Comment (LiKang6688) [reply src/object-properties.js]: aha, how about `Array<number>`? it is the way used in typescript.

**Comment by haxxmaxx** [comment]
shouldn't you generate the actual specs?

File: src/object-properties.js (hunk 2 above)
Comment (haxxmaxx) [reply src/object-properties.js]: I usually use `number[]` in jsdoc

Comment (LiKang6688) [reply src/object-properties.js]: aha, jsdoc3. https://jsdoc.app/tags-type.htm
Screenshot 2022-02-23 at 10 12 52
There is a dot between Array and <MyClass> which looks odd. I gonna use `number[]`.

Comment (LiKang6688) [reply src/object-properties.js]: Done

**Comment by LiKang6688** [comment]
> shouldn't you generate the actual specs?

No. Generating the actual specs fails the CI. I think that the actual specs can be generated when we bump the version.

**Comment by haxxmaxx** [comment]
Isn't that weird? Better to update it as soon as we can?

**Comment by LiKang6688** [comment]
> Isn't that weird? Better to update it as soon as we can?

I think it is correct that specs are not updated when did not update the version.

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**